cmake/AwsGetCmdArguments.cmake

```diff
@@ -0,0 +1,49 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+# Get list of variables provided via command line arguments (i.e. -DVARIABLE=VALUE). This function iterates over all
+# variables and determines if they're provided via command line or not by checking their help strings. CMake sets help
+# strings for variables provided via command line to the "No help, variable specified on the command line." phrase since
+# at least v3.0.
+# Via https://cmake.org/pipermail/cmake/2018-January/067002.html
```

Comment:
IMO:
- Move this to `AwsPrebuildDependency.cmake`
- name it something like `aws_get_cmd_args_for_prebuild_dependency()`
- don't bother taking arguments, just hardcode the CMAKE_INSTALL_PATH exception

You've done great working getting this to work, and it's noble to try and make it a generic helper, but let's be honest, all this complexity is an abomination, hoisted upon us because we can't use [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) from CMake 3.11, because we're stuck on 3.9 until we can drop support for RHEL5 (very extremely likely next year)

Reply:
Well, my rant about how FetchContent would solve all our problems was wrong.
Turns out FetchContent won't "prebuild" AWS-LC before configuring S2n
Maybe you explained this to me recently, and then I forgot?